test/renate/benchmark/test_experimentation_config.py

```diff
@@ -363,3 +365,17 @@ def test_loss_fn_returns_correct_reduction_type():
 def test_metrics_fn_contains_accuracy():
     assert isinstance(metrics_fn(num_outputs=2)["accuracy"], MulticlassAccuracy)
     assert isinstance(metrics_fn(num_outputs=10)["accuracy"], MulticlassAccuracy)
+
+
+def test_optimizer_fn():
+    expected_learning_rate = 0.12
+    partial_optimizer = optimizer_fn(
+        optimizer="AdamW", learning_rate=expected_learning_rate, weight_decay=0.1
+    )
+    optimizer: AdamW = partial_optimizer(Linear(10, 10).parameters())
+    assert isinstance(optimizer, AdamW)
+    assert optimizer.defaults["lr"] == expected_learning_rate
+
+
+def test_optimizer_fn_unknown_optimizer():
+    assert optimizer_fn(optimizer="UNKNOWN_OPTIMIZER", learning_rate=0.1, weight_decay=1) is None
```

Comment:
What does this test aim at accomplishing? Would throwing an error be better?

Reply:
similar to above. returning `None` is fine. This will be the case when using SGD or Adam. Raising an error in `optimizer_fn` should not happen. If the input is wrong, it will eventually be triggered at a different part of the code.